src/pkg/variables/templates.go

```diff
@@ -0,0 +1,155 @@
+
+	fileScanner := bufio.NewScanner(textFile)
+
+	// Set the buffer to 1 MiB to handle long lines (i.e. base64 text in a secret)
+	// 1 MiB is around the documented maximum size for secrets and configmaps
+	const maxCapacity = 1024 * 1024
+	buf := make([]byte, maxCapacity)
+	fileScanner.Buffer(buf, maxCapacity)
+
+	// Set the scanner to split on new lines
+	fileScanner.Split(bufio.ScanLines)
+
+	text := ""
+
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+
+		for {
+			matches := regexTemplateLine.FindStringSubmatch(line)
+
+			// No template left on this line so move on
+			if len(matches) == 0 {
+				text += fmt.Sprintln(line)
+				break
+			}
+
+			preTemplate := matches[regexTemplateLine.SubexpIndex("preTemplate")]
+			templateKey := matches[regexTemplateLine.SubexpIndex("template")]
+
+			_, present := vc.deprecatedKeys[templateKey]
+			if present {
+				deprecationWarning := fmt.Sprintf("This Zarf Package uses a deprecated variable: '%s' changed to '%s'.  Please notify your package creator for an update.", templateKey, vc.deprecatedKeys[templateKey])
+				vc.logger.Warn(deprecationWarning)
+			}
+
+			template := templateMap[templateKey]
+
+			// Check if the template is nil (present), use the original templateKey if not (so that it is not replaced).
+			value := templateKey
+			if template != nil {
+				value = template.Value
+
+				// Check if the value is a file type and load the value contents from the file
+				if template.Type == FileVariableType && value != "" {
+					if isText, err := helpers.IsTextFile(value); err != nil || !isText {
+						nonTextWarning := fmt.Sprintf("Refusing to load a non-text file for templating %s", templateKey)
+						vc.logger.Warn(nonTextWarning)
+						line = matches[regexTemplateLine.SubexpIndex("postTemplate")]
+						continue
+					}
+
+					contents, err := os.ReadFile(value)
+					if err != nil {
+						unableToReadWarning := fmt.Sprintf("Unable to read file for templating - skipping: %s", err.Error())
+						vc.logger.Warn(unableToReadWarning)
+						line = matches[regexTemplateLine.SubexpIndex("postTemplate")]
+						continue
+					}
+
+					value = string(contents)
+				}
+
+				// Check if the value is autoIndented and add the correct spacing
+				if template.AutoIndent {
+					indent := fmt.Sprintf("\n%s", strings.Repeat(" ", len(preTemplate)))
+					value = strings.ReplaceAll(value, "\n", indent)
+				}
+			}
+
+			// Add the processed text and continue processing the line
+			text += fmt.Sprintf("%s%s", preTemplate, value)
+			line = matches[regexTemplateLine.SubexpIndex("postTemplate")]
+		}
+	}
+
+	textFile.Close()
+
+	return os.WriteFile(path, []byte(text), helpers.ReadWriteUser)
+}
+
+// ProcessYamlFilesInPath iterates over all yaml files in a given path and performs Zarf templating + image swapping.
+func (vc *VariableConfig) ProcessYamlFilesInPath(path string) ([]string, error) {
```

Comment:
Need to test this, but we may be able to get rid of this function. The only time it's used is in the helm post-renderer, where I believe we are only changing one file anyway

Reply:
This could be moved around, but this operation is needed because of how `manifests` get templated - they don't have values files to inject the ### stuff into the template so they are templated here in the post render.

Reply:
Yeah, but the manifests still end up in one chart.yaml right?

https://github.com/defenseunicorns/zarf/pull/2414/files#diff-5916bf23df76c7a23e4ae35785691e484869179d67d8e22e59727483f641c109R48-R63

Reply:
Ah yep - we can definitely do that.